vimeo/VimeoNetworking
# cam-254-oauth_appexchange: allow apps to exchange an existing app token

#### Ticket
[CAM-254](https://vimean.atlassian.net/browse/CAM-254)

#### Ticket Summary
The API team has created a new endpoint that allows an application to exchange an access toke that was issued for a different Vimeo app for a new access token that is issued to the calling application.  This will allow apps like Cameo and DayTripper to automatically log a user in using a token that has already been saved by the Vimeo app.  This endpoint is also a crucial component of the Cameo -> Vimeo networking switchover work.

#### Implementation Summary
I added a new method to Request+Authentication that creates the proper request for the oauth/appexchange endpoint.  A new method (appTokenExchange(accessToken:completion:)) was added to the AuthenticationController class for client applications to use when exchanging a token. 

#### How to Test
1) Log in with client credentials
2) call AuthenticationController.appTokenExchange and pass in a valid Vimeo access token.  The completion handler should return a valid VIMAccount containing a new access token for the calling application to use.
